app/components-react/windows/NotificationsAndNews.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import { Menu } from 'antd';
+import { useVuex } from 'components-react/hooks';
+import { ModalLayout } from 'components-react/shared/ModalLayout';
+import { Services } from 'components-react/service-provider';
+import News from './News';
+import Notifications from './Notifications';
+import { $t } from 'services/i18n';
+import styles from './NotificationsAndNews.m.less';
+
+export default function NotificationsAndNews() {
+  const { NotificationsService } = Services;
+
+  const tabNames = {
+    news: $t('News'),
+    notifications: $t('Notifications'),
+  };
+
+  const [selectedTab, setSelectedTab] = useState(tabNames.news);
+
+  const { notificationGroups } = useVuex(() => ({
+    notificationGroups: {
+      unread: NotificationsService.views.getUnread(),
+    },
+  }));
+
+  useEffect(() => {
+    if (!!notificationGroups && notificationGroups.unread.length > 0) {
+      setSelectedTab(tabNames.notifications);
+    }
+  }, [notificationGroups]);
```

Comment:
This will probably no longer be an issue when you switch to using the `<Tabs>` component, but in general I don't think this is a good use of an effect.  Unless I'm missing something, the main purpose of this effect is just to set the tab we appear on when the window first opens.  That's much better done when initializing the state rather than afterwards with an effect.  Since there's async operations to get this information, it can easily be computed during initial render.

Reply:
Oh, ok thanks! I think it's something I picked up working with Redux, but also very good to know going forward. Thanks for the clarification!